File: frontend/ios/Hive/Hive/UtilityBelt.swift
//
//  UtilityBelt.swift
//  Hive
//
//  Created by Chuck Onwuzuruike on 10/4/18.
//  Copyright © 2018 Chuck Onwuzuruike. All rights reserved.
//

import Foundation

class UtilityBelt {
    
    enum UtilityBeltError: Error {
        case expectedTrueStatement
    }
    
    public static func checkStringNotEmpty(str: String) {
        trueOrFail(truth: !str.isEmpty)
    }
    
    public static func trueOrFail(truth: Bool) {
        if (!truth) {
            fatalError("Expected true statement.")
        }
    }
    
    public static func pluralOrSingular(num: Int, word: String) -> String {
        if num == 1 {
            return String(word.dropLast())
        }
        return word
    }
    
    public static func isValidEmail(email: String) -> Bool {
        return !email.contains(" ") && email.contains("@") && email.contains(".")
    }
    
    public static func isValidUsername(username: String) -> Bool {
        return !username.isEmpty && !username.contains(where: {
            return !$0.isLetter && !$0.isNumber
        })
    }
}

File: frontend/android/app/src/main/java/coloredcoded/hive/client/UtilityBelt.java
package coloredcoded.hive.client;

public class UtilityBelt {
	
	public static void checkStringNotEmpty(String str) {
		checkNotNull(str);
		trueOrFail(!str.isEmpty());
	}

	public static void checkNotNull(Object obj) {
		trueOrFail(obj != null);
	}

	public static void trueOrFail(boolean truth) {
		if (!truth) {
			throw new IllegalArgumentException();
		}
	}

	public static String pluralOrSingular(int num, String word) {
		if (num == 1) {
			return word.substring(0, word.length()-1);
		}
		return word;
	}

}
